File: http patch/Scala/fundservice/fund/src/main/scala/fund/NoPattern.scala
package fund

import fund.DomainModel._
import fund.PatchInstructionService.{patchFundName, patchFundType}
import fund.common.Pipeline.Result
import fund.common.SupportException

import scala.collection.mutable.ListBuffer

object NoPattern {

  def patchFund(id: Int, instructions: List[PatchFundDto], get: ((Int) => Fund), update: ((Fund) => Result[Fund])): Unit = {
    val errors = new ListBuffer[String]
    var currentData = get(id)
    var patchFundNameResult = Result[Fund](data = null)

    instructions.foreach(instruction => {
      instruction match {
        case instruction if instruction.path == "fundType" => {
          val patchFundTypeResult = patchFundType(currentData, instruction)(currentData)
          if (!patchFundTypeResult.success())
            errors += patchFundTypeResult.error

          currentData = patchFundTypeResult.data
        }
        case instruction if instruction.path == "fundName" => {
          patchFundNameResult = patchFundName(currentData, instruction)(currentData)
          if (!patchFundNameResult.success())
            errors += patchFundNameResult.error

          currentData = patchFundNameResult.data
        }
      }
    })

    val publishUpdateResult = publishUpdate(currentData, println)(currentData);
    if (!publishUpdateResult.success())
      errors += publishUpdateResult.error

    val validateNameResult = validateFundName(currentData)
    if (!validateNameResult.success())
      errors += validateNameResult.error

    currentData = validateNameResult.data

    val validateFundTypeResult = validateFundType(currentData)
    if (!validateFundTypeResult.success())
      errors += validateFundTypeResult.error

    errors.length > 0 match {
      case _ => update(currentData)
      case _ => throw new SupportException(errors.mkString("<br/>"))
    }
  }
}

File: http patch/python/fund/no_pattern.py
from domain_model import validate_fund_type, validate_fund_name, publish_update
from patch_instruction_service import patch_fund_type, FUND_TYPE_KEY, FUND_NAME_KEY, patch_fund_name


def patch_fund(fund_id, instructions, get, update):
    current_data = get(fund_id)

    for instruction in instructions:
        if instruction.path == FUND_TYPE_KEY:
            patch_fund_type_result = patch_fund_type(instruction)(current_data)
            if not patch_fund_type_result.success():
                return patch_fund_type_result.error
            current_data = patch_fund_type_result.data

        if instruction.path == FUND_NAME_KEY:
            patch_fund_name_result = patch_fund_name(instruction)(current_data)
            if not patch_fund_name_result.success():
                return patch_fund_name_result.error
            current_data = patch_fund_name_result.data

    validate_fund_type_result = validate_fund_type(current_data)
    if not validate_fund_type_result.success():
        return validate_fund_type_result.error
    current_data = validate_fund_type_result.data

    validate_fund_name_result = validate_fund_name(current_data)
    if not validate_fund_name_result.success():
        return validate_fund_name_result.error
    current_data = validate_fund_name_result.data

    publish_update_result = publish_update(current_data.fund_id, print)(current_data)
    if not publish_update_result.success():
        return publish_update_result.error
    current_data = publish_update_result.data

    update(current_data)
